target/linux/ar71xx/base-files/lib/ar71xx.sh

```diff
@@ -393,6 +393,154 @@ tplink_pharos_v2_get_model_string() {
 	dd if=$part iflag=skip_bytes bs=64 skip=4360 count=1 2>/dev/null | tr -d '\r\0' | head -n 1
 }
 
+mikrotik_board_detect() {
+	case "$machine" in
```

Comment:
I believe this is missing a `local` statement here and you need to pass the argument when you call that routine ($machine is local to the other function). If you want to be extra pedantic you could remove the already matched "MikroTik" part ;)

Reply:
I tested this script and "local machine" is no needed. If function 'ar71xx_board_detect' use function 'mikrotik_board_detect' then all local variable in 'ar71xx_board_detect' is used as "global" in 'mikrotik_board_detect'.
Mikrotik function set var "name" and ar71xx continue script.

Reply:
Ah right, that makes sense indeed. But then you’re relying on the parent function not renaming or deleting that variable. It might be ok in this context though.